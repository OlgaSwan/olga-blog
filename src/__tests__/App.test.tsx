import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'
import Component from '../app/component'

test('Renders the test component', () => {
  render(<Component />)

  expect(screen.getByTitle('heading')).toHaveTextContent('Component Heading')
})
test('main testing', () => {
  render(<Component />)

  const resetBtn = screen.getByText('Reset')
  const input = screen.getByPlaceholderText('Enter text')
  expect(resetBtn).toBeTruthy()
  expect(resetBtn).not.toBeDisabled()
  userEvent.click(resetBtn)
  expect(input).toHaveValue('')
})

test('footer testing', () => {
  render(<Component />)
  expect(screen.getByTitle('footering')).toBeTruthy()
  expect(screen.getByPlaceholderText('Disabled')).toBeDisabled()
})
