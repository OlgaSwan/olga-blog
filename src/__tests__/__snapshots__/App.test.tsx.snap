// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component main testing 1`] = `
<button>
  Reset
</button>
`;

exports[`Testing component main testing 2`] = `
<input
  placeholder="Enter text"
  type="text"
  value=""
/>
`;
